src/modules/shortcut.ts

```diff
@@ -0,0 +1,202 @@
+		if (LINK_REGEX.exec(description)?.[0] === description)
+			data = {
+				...base,
+				content: description,
+			};
+		else if (description)
+			data = {
+				...base,
+				title,
+				description,
+				color: BLOCKQUOTE_GREY_NUMBER,
+			};
+		else if (referencedEmbed)
+			data = {
+				...base,
+				title:
+					referencedEmbed.title && title
+						? `${title}: ${referencedEmbed.title}`
+						: referencedEmbed.title || title,
+				description: referencedEmbed.description,
+				color: referencedEmbed.color,
+				image: referencedEmbed.image?.url,
+				url: referencedEmbed.url,
+			};
+
+		if (!data)
+			return await sendWithMessageOwnership(
+				msg,
+				':x: Cannot generate a shortcut from that message',
+			);
+
+		await existingShortcut?.remove();
+		await Shortcut.create(data).save();
+		await sendWithMessageOwnership(
+			msg,
+			`:white_check_mark: ${
+				existingShortcut ? 'Edited' : 'Created'
+			} shortcut \`${id}\``,
+		);
+	}
+
+	private async getShortcut(id: string) {
+		return await Shortcut.findOne(id);
+	}
+
+	@command({
+		aliases: ['ls'],
+		description: 'Shortcut: List the most used shortcuts',
+	})
+	async listShortcuts(msg: Message) {
+		const shortcuts: Shortcut[] = await Shortcut.createQueryBuilder()
+			.select(['id', 'uses'])
+			.orderBy('uses', 'DESC')
+			.limit(10)
+			.getRawMany();
+		await sendWithMessageOwnership(msg, {
+			embed: new MessageEmbed()
+				.setTitle('Top 10 Shortcuts')
+				.setColor(TS_BLUE)
+				.setDescription(
+					shortcuts.map(
+						shortcut =>
+							`- \`${shortcut.id}\` with **${shortcut.uses}** uses.`,
+					),
+				),
+		});
+	}
+
+	@command({
+		aliases: ['lsm'],
+		description: 'Shortcut: List shortcuts owned by you',
+	})
+	async listMyShortcuts(msg: Message) {
+		const shortcuts: Shortcut[] = await Shortcut.createQueryBuilder()
+			.select(['id', 'owner', 'uses'])
+			.where('owner = :userId')
+			.orderBy('id', 'DESC')
+			.setParameters({ userId: msg.author.id })
+			.getRawMany();
+		await sendWithMessageOwnership(msg, {
+			embed: new MessageEmbed()
+				.setTitle('Your Shortcuts')
+				.setColor(TS_BLUE)
+				.setDescription(
+					shortcuts.map(shortcut => `- \`${shortcut.id}\``),
+				),
+		});
+	}
+
+	@command({
+		description: 'Shortcut: Delete a shortcut you own',
+	})
+	async deleteShortcut(msg: Message, id: string) {
+		const shortcut = await this.getShortcut(id);
+		if (!shortcut)
+			return await sendWithMessageOwnership(
+				msg,
+				':x: No shortcut with that id',
+			);
+		if (shortcut.owner !== msg.author.id)
+			return await sendWithMessageOwnership(
+				msg,
+				":x: Cannot delete another's shortcut",
+			);
+		await shortcut.remove();
+		sendWithMessageOwnership(msg, ':white_check_mark: Deleted shortcut');
+	}
+
+	private getId(user: User, name: string) {
+		const sanitize = (part: string) =>
+			part.toLowerCase().replace(/[^\w-]/g, '');
+		return `${sanitize(user.username)}:${sanitize(name)}`;
```

Comment:
What happeneds if the user changes their username? Does that break anything? Hiding this logic at the bottom of the file might cause future contributors to trust the returned value in unsafe ways.

Reply:
It won't break it; it also stores the owner's id and checks based off of that.